src/ClientData/include/ClientData/FunctionInfo.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CLIENT_DATA_FUNCTION_INFO_H_
+#define CLIENT_DATA_FUNCTION_INFO_H_
+
+#include <optional>
+#include <string>
+#include <utility>
+
+#include "GrpcProtos/symbol.pb.h"
+
+namespace orbit_client_data {
+
+class ModuleData;
+class ProcessData;
+
+// This class is used on the client to represent a function, containing a name, module path and
+// build ID, address and size.
+class FunctionInfo {
+ public:
+  FunctionInfo() = delete;
```

Comment:
Nit: This is not needed. By defining a custom constructor the auto generation of a default constructor gets automatically disabled.